src/platforms/web/runtime/modules/events.js

```diff
@@ -44,6 +44,14 @@ function createOnceHandler (event, handler, capture) {
 // safe to exclude.
 const useMicrotaskFix = isUsingMicroTask && !(isFF && Number(isFF[1]) <= 53)
 
+// #10366: CEP <= 9.3.x has a buggy Event.timeStamp implementation. While the
+// issue is restricted to macOS, the fix is OS-agnostic to keep behavioral
+// differences to a minimum.
+const isCEP93orEarlier = isCEP && ((maxBadMajor, maxBadMinor) => {
+  const version = JSON.parse(window.__adobe_cep__.getCurrentApiVersion())
```

Comment:
are we sure this function exist in **all** versions of adobe_cep?

Reply:
If it exist in all version till today, I think we should check if the function exists and if it doesn't return false to prevent breaking changes with adobe_cep api in the future

Reply:
It has existed in all CEP versions since its inception in [version 4.x ](https://github.com/Adobe-CEP/CEP-Resources/blob/5880e20b7cb2dc09e6b226f15a33165a08c0d831/CEP_4.x/CSInterface.js#L812)

To my knowledge, Adobe has never removed functions from their CEP API, even deprecated ones. However, a check for the function is still a good idea.

Reply:
>are we sure this function exist in all versions of adobe_cep?

Not 100%, no. The table shows that the earliest version of  **HTML**-based CEP was "CEP 4". Adobe has an extraordinarily bad habit of not publishing details of the CEP minor versions for their apps so it is entirely possible that "version 4" means "version 4.2." That said, the opposite is also possible.

>If it exist in all version till today, I think we should check if the function exists and if it doesn't return false to prevent breaking changes with adobe_cep api in the future

@bbb999 might be able to shed more light on this but my understanding is that the upcoming CEP 9.4.x is likely to be the last version of CEP. Adobe is _beginning_ to transition their apps to a new technology (known as UXP) and are starting to phase out CEP. There may be bugfix releases but I'd be willing to bet that we won't see any core API changes before CEP's final death knell.

>However, a check for the function is still a good idea.

Agreed. I can make that change for sure.

Reply:
Adobe has made no announcements regarding [above], but Eric's speculation is entirely plausible. :)

Reply:
@bbb999 I'm going to add a check for the existence of the `getCurrentApiVersion` function. It's non-existence, however, can mean one of two things:

1. This is an archaic version of CEP that pre-dates that function.
2. This is a future version of CEP that redefined that function as something else.

Given the trajectory of CEP over its lifetime, it seems that option 2 is far, far less likely than option 1. Would it be safe to assume, then, that if:

```js
typeof __adobe_cep__.getCurrentApiVersion !== 'function'
```

then we're more likely to be working with an archaic version of CEP than an as-yet-unreleased one?

Reply:
Spoke with @bbb999 in another forum (Adobe Devs community Slack) and the response was:

>[Option 1], definitely.